src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -500,10 +510,14 @@ bool serdata_to_sample(
 {
   (void)bufptr;
   (void)buflim;
-  auto ptr = static_cast<const ddscxx_serdata<T>*>(dcmn);
 
-  auto& msg = *static_cast<T*>(sample);
-  return deserialize_sample_from_buffer(ptr->data(), ptr->size(), msg);
+  auto typed_sample_ptr = static_cast<T*>(sample);
+  // cast away const, with the reasoning that we don't modify the underlying ddsi_serdata which
+  // is actually const, we only modify the ddscxx_serdata non const contents
+  auto d = const_cast<ddscxx_serdata<T>*>(static_cast<const ddscxx_serdata<T>*>(dcmn));
```

Comment:
I thought of using some free functions to get the sample from `serdata` (either from `iox_chunk` or actually deserialize from the buffer) instead of using `const_cast`. However, with that approach, we will need always to deserialize since we don't store the sample as part of `ddscxx_serdata`. So I took the approach to cast away the const on `ddscxx_serdata` and then modify the non-const fields in `ddscxx_serdata`. 

There is also another place in the code here https://github.com/eclipse-cyclonedds/cyclonedds-cxx/blob/653e97e2ba5bc3a81ecf9c4edba3493c0f08bdb8/src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp#L516 where this is done, so I guess this should be okay.

CC @eboasson @reicheratwork

Reply:
This looks like it does what it's supposed to and with fewer lines of code and without the complete deserialization step that deserialize_sample_from_buffer introduced in all cases, here the deserialization will only be done if the sample is not yet deserialized in the ddscxx_serdata entity. So this should be faster.

The only issue i see is it not checking the result of any deserialization. While it is a small probability, something could have gone wrong during deserialization and then the getT function will return a nullptr.
I suggest the following small change from:
```c++
 *typed_sample_ptr = *d->getT();

  return true;
```
to:
```c++
  auto t_ptr = d->getT();
  if (!t_ptr)
    return false;

  *typed_sample_ptr = *t_ptr;
  return true;
```

Reply:
@reicheratwork Good point, fixed it.